core/block.js

```diff
@@ -964,7 +965,9 @@ Blockly.Block.prototype.getFieldValue = function(name) {
  */
 Blockly.Block.prototype.setFieldValue = function(newValue, name) {
   var field = this.getField(name);
-  goog.asserts.assertObject(field, 'Field "%s" not found.', name);
+  if (!field) {
```

Comment:
Should this be -- `if (!field || typeof field != 'object')`?

Reply:
this.getField returns a Field object, or null.  No other return values are possible.  There's no need to check for anything else.